### PR TITLE
HealthPlan add and display Meal Plan enhancement

### PR DESCRIPTION
**Pull request by AlecNgZW**
some minor UI implementation to hide and show the mealPlan list on the browser panel

### CODE REVIEW

File: src/main/java/seedu/souschef/logic/commands/ShowHealthPlanDetailsCommand.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    public static final String COMMAND_WORD = "showDetails";
+
+    public static final String MESSAGE_SUCCESS = "showing details of plan %1$s.";
+
+    private final Ui ui;
+
+    private final String index;
+
+
+    public ShowHealthPlanDetailsCommand(Ui ui, String index) {
```
Comment (junrong09) [inline src/main/java/seedu/souschef/logic/commands/ShowHealthPlanDetailsCommand.java]: Please make us of event center to subscribe to change in event.

File: src/main/java/seedu/souschef/logic/commands/AddMealHealthPlanCommand.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+
+    public static final String COMMAND_WORD = "addDay";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Adds the day identified by the index number used in the "
+            + "displayed meal plan list to the specified health plan\n"
+            + "Parameters: p/PLANINDEX d/DAYINDEX(must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " p/1 d/1";
+
+    public static final String MESSAGE_SUCCESS = "added day %2$s to plan %1$s";
```
Comment (junrong09) [inline src/main/java/seedu/souschef/logic/commands/AddMealHealthPlanCommand.java]: Capitalise the first word of the message.

File: src/main/java/seedu/souschef/logic/commands/AddMealHealthPlanCommand.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    private Model<HealthPlan> model;
+
+    private ArgumentMultimap argMultimap;
+
+    private int index;
+
+    public AddMealHealthPlanCommand(Model healthPlanModel, Model mealPlanModel, String args) throws ParseException {
+        this.model = healthPlanModel;
+
+        argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_PLAN, PREFIX_DURATION);
```
Comment (junrong09) [inline src/main/java/seedu/souschef/logic/commands/AddMealHealthPlanCommand.java]: Do perform parsing in ur AddMealHealthPlanCommandParser.

File: src/main/java/seedu/souschef/logic/commands/DeleteMealHealthPlanCommand.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+    public static final String COMMAND_WORD = "deleteDay";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Removes the day identified by the index number used in the "
+            + "displayed meal plan list to the specified health plan\n"
+            + "Parameters: p/PLANINDEX d/DAYINDEX(must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " p/1 d/1";
+
+    public static final String MESSAGE_SUCCESS = "removed day %2$s from plan %1$s";
```
Comment (junrong09) [inline src/main/java/seedu/souschef/logic/commands/DeleteMealHealthPlanCommand.java]: Capitalise the first word

File: src/main/java/seedu/souschef/logic/commands/DeleteMealHealthPlanCommand.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+    private int indexToRemove;
+
+    private int index;
+
+    public DeleteMealHealthPlanCommand(Model healthPlanModel, String args) throws ParseException {
+        this.model = healthPlanModel;
+
+
+        argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_PLAN, PREFIX_DURATION);
```
Comment (junrong09) [inline src/main/java/seedu/souschef/logic/commands/DeleteMealHealthPlanCommand.java]: Please perform parsing in ur command parser.

File: src/main/java/seedu/souschef/logic/commands/DisplayMealPlanCommand.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * command extension to show all meal plans logged into the system
+ */
+public class DisplayMealPlanCommand<T extends UniqueType> extends Command {
+
+    public static final String COMMAND_WORD_SHOW = "showMeal";
+    public static final String COMMAND_WORD_HIDE = "hideMeal";
+
+    public static final String MESSAGE_SUCCESS_SHOW = "showing all meal plans.";
+    public static final String MESSAGE_SUCCESS_HIDE = "hide meal plans.";
```
Comment (junrong09) [inline src/main/java/seedu/souschef/logic/commands/DisplayMealPlanCommand.java]: Capitalisation

File: src/main/java/seedu/souschef/logic/commands/ShowHealthPlanDetailsCommand.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+
+/**
+ * command extension to show details for indexed healthplan
+ */
+public class ShowHealthPlanDetailsCommand<T extends UniqueType> extends Command {
+
+    public static final String COMMAND_WORD = "showDetails";
+
+    public static final String MESSAGE_SUCCESS = "showing details of plan %1$s.";
```
Comment (junrong09) [inline src/main/java/seedu/souschef/logic/commands/ShowHealthPlanDetailsCommand.java]: Capitalisation

File: src/main/java/seedu/souschef/logic/parser/AppContentParser.java [hunk 8]
```diff
@@ -55,15 +56,18 @@ public Command parseCommand(ModelSet modelSet, String userInput, CommandHistory
         case INGREDIENT:
             setFeatureStorage(storage, Context.INGREDIENT);
             return new IngredientParser().parseCommand(modelSet.getIngredientModel(), userInput);
+
         case HEALTH_PLAN:
             setFeatureStorage(storage, Context.HEALTH_PLAN);
-            return new HealthPlanParser().parseCommand(modelSet.getHealthPlanModel(), userInput);
+            return new HealthPlanParser().parseCommand(modelSet.getHealthPlanModel(), modelSet.getMealPlannerModel(),
+                    userInput, ui);
```
Comment (junrong09) [inline src/main/java/seedu/souschef/logic/parser/AppContentParser.java]: With event, do you still need UI as parameter?

File: src/main/java/seedu/souschef/logic/parser/contextparser/HealthPlanParser.java [hunk 9]
```diff
@@ -34,12 +39,13 @@
...
      *
-     * @param model
+     * @param healthPlanModel
      * @param userInput full user input string
      * @return the command based on the user input
      * @throws ParseException if the user input does not conform the expected format
      */
-    public Command<HealthPlan> parseCommand(Model model, String userInput) throws ParseException {
+    public Command<HealthPlan> parseCommand(Model healthPlanModel, Model mealPlanModel,
+                                            String userInput, Ui ui) throws ParseException {
```
Comment (junrong09) [inline src/main/java/seedu/souschef/logic/parser/contextparser/HealthPlanParser.java]: Can UI be removed as parameter?

File: src/main/java/seedu/souschef/logic/parser/contextparser/UniversalParser.java [hunk 10]
```diff
@@ -36,23 +38,33 @@
     public Command parseCommand(CommandHistory history, String userInput, Ui ui) throws ParseException {
         final String commandWord = userInput.substring(1);
         switch (commandWord) {
+
+
         case RecipeParser.COMMAND_WORD:
+            EventsCenter.getInstance().post(new BrowserUiChangedEvent("hide"));
```
Comment (junrong09) [inline src/main/java/seedu/souschef/logic/parser/contextparser/UniversalParser.java]: Perform the posting of hide event in the ui.switchToXXXListPanel() method. Same for other context switch.

**Review by junrong09 — CHANGES_REQUESTED** [review]
Capitalise all the first word of ur command success messages.
Create separate classes for parsing ur command, try not to do it within ur command class itself.
Other minor reviews are as stated in the individual review comments.

File: src/main/java/seedu/souschef/logic/parser/ParserUtil.java [hunk 11]
```diff
@@ -220,4 +220,6 @@ public static Scheme parseScheme(String scheme) throws ParseException {
 
 
 
+
```
Comment (BennTay) [inline src/main/java/seedu/souschef/logic/parser/ParserUtil.java]: Please remove excess newlines

File: src/main/java/seedu/souschef/logic/commands/AddMealHealthPlanCommand.java [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+import seedu.souschef.logic.History;
+import seedu.souschef.model.Model;
+import seedu.souschef.model.UniqueType;
+import seedu.souschef.model.healthplan.HealthPlan;
+import seedu.souschef.model.planner.Day;
+
+/**
+ * command to add indexed day into index healthplan
+ */
+public class AddMealHealthPlanCommand <T extends UniqueType> extends Command {
```
Comment (barkhyehyeon) [inline src/main/java/seedu/souschef/logic/commands/AddMealHealthPlanCommand.java:17]: I guess this command doesn't have to be generic, but just leaving it as it is would also be fine!

**Review by barkhyehyeon — DISMISSED** [review]
Good implementation as always. Approve PR.

**Review by junrong09 — APPROVED** [review]
Approve PR. Comprehensive enhancement of health plan feature.

**Pull request merged**